pytest/lib/mocknet.py

```diff
@@ -1296,41 +1229,48 @@ def wait_genesis_updater_done(node, done_filename):
 # list_validators but assumes that the node can still be starting and expects
 # connection errors and the data not being available.
 def wait_node_up(node):
+
+    class NeardResponseException(Exception):
+        pass
+
     msg = f'Waiting for node {node.instance_name} to start'
     logger.info(msg)
     attempt = 0
     while True:
         try:
+            result = node.machine.run('ps -A | grep neard')
+            if result.returncode != 0:
+                raise Exception(f'{msg} - failed. The neard process crashed.')
+
             response = node.get_validators()
 
             if 'error' in response:
-                raise Exception(
-                    f'Received error from node {node.instance_name}. Error: {response["error"]}'
-                )
+                raise NeardResponseException('error')
```

Comment:
not a big deal, leave as is if you like. But would it be more direct to just proceed normally here without raising an exception? like:

```diff
diff --git a/pytest/lib/mocknet.py b/pytest/lib/mocknet.py
index 768678279..699468edd 100644
--- a/pytest/lib/mocknet.py
+++ b/pytest/lib/mocknet.py
@@ -1229,10 +1229,6 @@ def wait_genesis_updater_done(node, done_filename):
 # list_validators but assumes that the node can still be starting and expects
 # connection errors and the data not being available.
 def wait_node_up(node):
-
-    class NeardResponseException(Exception):
-        pass
-
     msg = f'Waiting for node {node.instance_name} to start'
     logger.info(msg)
     attempt = 0
@@ -1244,19 +1240,14 @@ def wait_node_up(node):
 
             response = node.get_validators()
 
-            if 'error' in response:
-                raise NeardResponseException('error')
-            if 'result' not in response:
-                raise NeardResponseException('result missing')
-
-            logger.info(f'{msg} - done.')
-            return True
+            if 'error' not in response and 'result' in response:
+                logger.info(f'{msg} - done.')
+                return True
+            attempt += 1
+            logger.info(f'{msg}, validators RPC not ready')
         except (ConnectionRefusedError, requests.exceptions.ConnectionError):
             attempt += 1
             logger.info(f'{msg}, attempt {attempt} connection refused.')
-        except NeardResponseException:
-            attempt += 1
-            logger.info(f'{msg}, attempt {attempt} error response.')
         time.sleep(30)
  
```

Just feels like the general pattern of:

```python
try:
  if something():
    do_something()
  else:
     raise NotSomethingException
except NotSomethingException as e:
  do_other_thing()
```

could be simplified/flattened to more direct code